core/snippets/snippets_insert.py

```diff
@@ -0,0 +1,42 @@
+from talon import Module, actions
+
+from .snippet_types import Snippet
+from .snippets_insert_raw_text import insert_snippet_raw_text
+
+mod = Module()
+
+
+@mod.action_class
+class Actions:
+    def insert_snippet(body: str):
+        """Insert snippet"""
+        insert_snippet_raw_text(body)
+
+    def insert_snippet_by_name(name: str, substitutions: dict[str, str] = None):
+        """Insert snippet <name>"""
+        snippet: Snippet = actions.user.get_snippet(name)
+        body = snippet.body
+
+        if substitutions:
+            for k, v in substitutions.items():
+                body = body.replace(f"${k}", v)
+
+        actions.user.insert_snippet(body)
+
+    def insert_snippet_by_name_with_phrase(name: str, phrase: str):
+        """Insert snippet <name> with phrase <phrase>"""
+        snippet: Snippet = actions.user.get_snippet(name)
+        substitutions = {}
+
+        for variable in snippet.variables:
+            if variable.insertionFormatters is not None:
+                formatters = ",".join(variable.insertionFormatters)
+                formatted_phrase = actions.user.formatted_text(phrase, formatters)
+                substitutions[variable.name] = formatted_phrase
+
+        actions.user.insert_snippet_by_name(name, substitutions)
+
+    def code_insert_snippet_by_name(name: str, substitutions: dict[str, str] = None):
```

Comment:
This is used if you programmatically want to insert a specific snippet for the current programming language. `ifStatement` => `python.ifStatement`

Reply:
Also, is this action the one we would use if we started migrating our community language support to use snippets?

Reply:
Yes the other ones are global. There are snippets that are not for code and in those cases you would probably call the global functions directly. 
The reason for it being prefixed with code is because it is for code only.

Exactly. Here you have my python function implementation. `"private function my funk"`
https://github.com/AndreasArvidsson/andreas-talon/blob/4667f8f894d272d78e5b272a32694ff516ab204d/languages/python/python.py#L127-L131

Reply:
But in theory the global ids should work in any language even if you prefix them, due to the way that you always fallback to `_`, no? I like the idea of exposing one action that just works and handles context overrides properly. From what I can tell you've already done the hard work

Reply:
No doesn't work like that. The language scoped snippets need a specified language.

Reply:
Yes but I'm saying you should be able to use language prefix whether or not the scope is global, so we could have an action that just always prepends language prefix

Reply:
Isnt that what this action is doing?

Reply:
Updated according to discussion in meet up